cli/tools/run.rs

```diff
@@ -38,10 +38,17 @@ To grant permissions, set them before the script argument. For example:
     return run_with_watch(flags, run_flags.script).await;
   }
 
+  let main_module = if NpmPackageReference::from_str(&run_flags.script).is_ok()
+  {
+    ModuleSpecifier::parse(&run_flags.script)?
+  } else {
+    resolve_url_or_path(&run_flags.script)?
+  };
+
   // TODO(bartlomieju): actually I think it will also fail if there's an import
   // map specified and bare specifier is used on the command line - this should
   // probably call `ProcState::resolve` instead
-  let ps = ProcState::build(flags).await?;
+  let ps = ProcState::build_with_main(flags, main_module.clone()).await?;
```

Comment:
Could this second argument be derived from `flags`? Do we need to have a new `build` method and separate argument? Maybe we just need a new method on `flags` called `main_module()` so we can reuse the code above in multiple places?

Reply:
`main_module()` won't work for all relevant subcommands. `eval_command()` creates its main module on the fly. Moving that "fake" main module name into `main_module()` would be trivial, but confusing. From a developer's point of view, I'd expect `flags.main_module()` to return `None` if `deno eval 123` is invoked. I understand the main module of eval to be an implementation detail of eval which should not be known to `Flags` in my opinion.

Adding `build_with_main()` wasn't my first choice. Suggestions are welcome

Reply:
Changed by adding `CliOptions::resolve_main_module()`. Isn't as bad as I thought